orc8r/cloud/go/obsidian/swagger/registry_test.go

```diff
@@ -0,0 +1,58 @@
+/*
+ Copyright 2020 The Magma Authors.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package swagger_test
+
+import (
+	"testing"
+
+	"magma/orc8r/cloud/go/obsidian/swagger"
+	"magma/orc8r/cloud/go/orc8r"
+	"magma/orc8r/cloud/go/test_utils"
+	"magma/orc8r/lib/go/registry"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRegistry_GetSpecServicers(t *testing.T) {
+	// Success with no registered servicers
+	servicers, err := swagger.GetSpecServicers()
+	assert.NoError(t, err)
+
+	assert.Empty(t, servicers)
+
+	// Success with some registered servicers
+	services := map[string]struct{}{
+		"test_spec_service1": {},
+		"test_spec_service2": {},
+	}
+	labels := map[string]string{
+		orc8r.SwaggerSpecLabel: "true",
+	}
+
+	// Clean up registry
+	defer registry.RemoveServicesWithLabel(orc8r.SwaggerSpecLabel)
+
+	for s := range services {
+		srv, lis := test_utils.NewTestOrchestratorService(t, orc8r.ModuleName, s, labels, nil)
+		go srv.RunTest(lis)
+	}
+
+	servicers, err = swagger.GetSpecServicers()
+	assert.NoError(t, err)
+
+	_, exists := services[servicers[0].GetService()]
```

Comment:
These checks are insufficient -- what if both returned servicers were `test_spec_service1`?

I remembered there's a good way to handle this actually: go back to the way you had it before, but use assert.ElementsMatch

Reply:
Good point, also thanks for the tip on the cleaner method.